service/storage_service.cc

```diff
@@ -649,6 +680,23 @@ topology_mutation_builder& topology_mutation_builder::set_global_topology_reques
     return *this;
 }
 
+template<ConvertibleToSstring S>
+topology_mutation_builder& topology_mutation_builder::add_enabled_features(const std::set<S>& features) {
+    auto cdef = _s->get_column_definition("enabled_features");
+    assert(cdef);
+    auto vtype = static_pointer_cast<const set_type_impl>(cdef->type)->get_elements_type();
+
+    collection_mutation_description cm;
+    cm.cells.reserve(features.size());
+
+    for (auto&& value : features) {
+        cm.cells.emplace_back(vtype->decompose(sstring(value)), atomic_cell::make_live(*bytes_type, _ts, bytes_view()));
+    }
+
+    _m.partition().static_row().apply(*cdef, cm.serialize(*cdef->type));
+    return *this;
+}
```

Comment:
I think those two function can mostly shared the code as well. The only serious difference is `row().apply()` vs `static_row().apply()`

Reply:
> I think those two function can mostly shared the code as well.

I'll try.

> The only serious difference is `row().apply()` vs `static_row().apply()`

The other important difference is that it adds features instead of overwriting them.